eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -817,6 +817,15 @@ class Channel(val nodeParams: NodeParams, wallet: EclairWallet, remoteNodeId: Pu
       val channelUpdate = Announcements.makeChannelUpdate(nodeParams.chainHash, nodeParams.privateKey, remoteNodeId, d.shortChannelId, d.channelUpdate.cltvExpiryDelta, d.channelUpdate.htlcMinimumMsat, d.channelUpdate.feeBaseMsat, d.channelUpdate.feeProportionalMillionths, enable = true)
       stay using store(d.copy(channelUpdate = channelUpdate))
 
+    case Event(CMD_UPDATE_RELAY_FEE(feeBaseMsat, feeProportionalMillionths), d: DATA_NORMAL) =>
+      // Command from API
+      log.debug(s"sending channel_update announcement (new relay fees $feeBaseMsat $feeProportionalMillionths)")
+      val channelUpdate = Announcements.makeChannelUpdate(nodeParams.chainHash, nodeParams.privateKey, remoteNodeId, d.shortChannelId, d.channelUpdate.cltvExpiryDelta, d.channelUpdate.htlcMinimumMsat, feeBaseMsat, feeProportionalMillionths, enable = true)
+      sender ! "ok"
+      forwarder ! d.channelUpdate
+      context.system.eventStream.publish(LocalChannelUpdate(self, d.commitments.channelId, d.shortChannelId, d.commitments.remoteParams.nodeId, d.channelAnnouncement, channelUpdate))
```

Comment:
This is already handled in the state transition and will result in duplicate events

Reply:
I will retest this. I had a 4 node regtest network running and was sure the updates were not getting propagated without this.
I even changed the 7 day announcement to 7 seconds to see if that helped.